client/control/components/CheckboxList.js

```diff
@@ -3,26 +3,61 @@ import React, { PropTypes as pt } from 'react';
 /**
  * Simple component which lists a bunch of checkboxes,
  * and emits input changes.
+ *
+ * @prop {Function} onInputChange
+ *   Fires when user has changed a checkbox, with
+ *   params (Item index, Checkbox 'checked' status)
+ *
+ * @prop {Array<Object>} options
+ *   List of options to be made into checkboxes, shaped as:
+ *   [{
+ *     label: 'Display label',
+ *     value: 'Checkbox value',
+ *     enabled: true || false, // boolean
+ *   }, ...]
  */
-
 export default class CheckboxList extends React.Component {
   static propTypes = {
     onInputChange: pt.func.isRequired,
     options: pt.array.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    // Cache of generated event handlers
+    this.handlerCache = {};
+  }
+
   /**
-   * Handler factory. Given an index,
+   * @param  {[type]} newProps [description]
+   * @return {[type]}          [description]
```

Comment:
Yeah, in retrospect we don't need the `componentWillReceiveProps` at all - the cache is keyed on column slug, so if the options update, the cache is still relevant.